src/main/java/lia/Storage.java

```diff
@@ -0,0 +1,108 @@
+package lia;
+
+import java.util.ArrayList;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+
+/**
+ * The Storage class handles the loading and saving of tasks from/to a file.
+ * It interacts with the TaskList and various task-related classes for file I/O operations.
+ */
+public class Storage {
+    private static final String FILE_NAME = "tasks.txt";
+    private static final String FOLDER_NAME = "data";
+    private static final String PROJECT_FOLDER = ".";
+    private static final Path FILE_PATH = Paths.get(PROJECT_FOLDER, FOLDER_NAME, FILE_NAME);
+
+    /**
+     * Constructs a Storage object, ensuring the existence of necessary directories.
+     * It creates the required directory structure for storing task data.
+     */
+    public Storage() {
+        try {
+            Files.createDirectories(FILE_PATH.getParent());
+        } catch (IOException e) {
+            System.out.println("Error creating directories: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Loads tasks from a file and returns them as an ArrayList.
+     *
+     * @return An ArrayList containing the loaded tasks.
+     * @throws LiaException If an error occurs during the loading process.
+     */
+    public ArrayList<Task> loadTasks() throws LiaException {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            Files.createDirectories(FILE_PATH.getParent());
+            try (BufferedReader br = Files.newBufferedReader(FILE_PATH)) {
+                String line;
+                while ((line = br.readLine()) != null) {
+                    String[] parts = line.split(" \\| ");
+                    String type = parts[0];
+                    boolean isDone = parts[1].equals("1");
+                    String description = parts[2];
+
+                    switch (type) {
```

Comment:
Might want to remove indentation for switch clauses (as per the coding conventions mentioned in this course)